src/main/java/org/folio/innreach/domain/service/impl/PatronInfoServiceImpl.java

```diff
@@ -118,6 +128,43 @@ private User findPatronUser(String visiblePatronId, String patronName) {
     return user;
   }
 
+  private String constructPublicIdQuery(VisiblePatronFieldConfiguration fieldConfig, String visiblePatronId) {
+    var fields = fieldConfig.getFields();
+    var checkCustomFields = fields.remove(USER_CUSTOM_FIELDS);
+    var fieldsString = fields.stream().map(VisiblePatronFieldConfiguration.VisiblePatronField::getValue)
+      .collect(Collectors.toList());
+
+    var query = String.join(QUERY_EXACT_DELIMITER, fieldsString);
+    query += QUERY_EXACT_DELIMITER;
+    if (checkCustomFields) {
+      query = addCustomFieldsToQuery(query, fieldConfig);
+    }
+    query = query.substring(0, query.length() - 4);
+
+    return String.format(query, visiblePatronId);
+  }
+
+  private String addCustomFieldsToQuery(String query, VisiblePatronFieldConfiguration fieldConfig) {
+    var customFields = fieldConfig.getUserCustomFieldIds().stream()
+      .map(customFieldsClient::getCustomField).collect(Collectors.toList());
+    var queryBuilder = new StringBuilder(query);
+    customFields.forEach(field -> {
+      switch (field.getType()) {
+        case RADIO_BUTTON:
+        case SINGLE_CHECKBOX:
+        case SINGLE_SELECT_DROPDOWN:
+        case TEXTBOX_SHORT:
+        case TEXTBOX_LONG:
+          queryBuilder.append("customFields.").append(field.getRefId()).append(QUERY_EXACT_DELIMITER);
+          break;
+        case MULTI_SELECT_DROPDOWN:
+          queryBuilder.append(QUERY_ANY_DELIMITER).append("customFields.").append(field.getRefId()).append(" or ");
+          break;
+      }
```

Comment:
Considering that we are implementing these configs similar to how it's done for Circulation settings, the only supported types should be: TEXTBOX_SHORT and TEXTBOX_LONG